tests/jballerina-unit-test/src/test/resources/test-src/expressions/mappingconstructor/spread_op_field_semantic_analysis_negative.bal

```diff
@@ -71,6 +71,66 @@ function testFieldWithNeverType() {
     Foo _ = {...rec5, ...rec6};
 }
 
+type Address record {
+    string street;
+};
+
+type Street record {|
+    string street;
+|};
+
+type Foz record {|
+    string s;
+    int i;
+    int j;
+    boolean b;
+    string...;
+|};
+
+function testSpreadOfOpenRecordToCreateClosedRecord() {
+    Address address1 = {street: "Main Street"};
+    record {|string street;|} _ = {...address1};
+
+    Address address2 = {street: "Pine Street", "number": 40};
+    record {|string street;|} _ = {...address2};
+
+    Address address3 = {street: "Maple street" };
+    Street _ = {...address3};
+
+    record {string street;} address4 = {street: "Jump Street"};
+    Street _ = {...address4};
+
+    record {string street;} address5 = {street: "Willow Street", "number": 60};
+    Street _ = {...address5};
+
+    record {string s;} foo1 = {s: "S"};
+    record {int i;} foo2 = {i: 2};
+    Foo _ = {...foo1, ...foo2};
+
+    record {|string s;|} foz1 = {s: "s"};
+    record {|int i;|} foz2 = {i: 1};
+    record {int j;} foz3 = {j: 2};
+    record {boolean b;} foz4 = {b: true};
+    Foz _ =  {...foz1, ...foz2, ...foz3, ...foz4};
+}
+
+type Country record {|
+    string name;
+    string continent;
+    string...;
+|};
+
+function testSpreadOpFieldOfMismatchingRestType() {
+    record {string name; string continent; int population;} country1 = {name: "China", continent: "Asia", population: 1400};
+    Country _ = {...country1};
+
+    record {string name; string continent;} country2 = {name: "Sri Lanka", continent: "Asia"};
+    Country _ = {...country2};
+
+    record {string name;} country3 = {name: "India"};
+    Country _ = {continent: "Asia", ...country3};
```

Comment:
I would add a test(s) for different rest field types instead. 

implicit vs explicit
explicit vs explicit
explicit vs implicit

Reply:
Addressed [here](https://github.com/ballerina-platform/ballerina-lang/pull/41784/commits/23e0d92e220c42c9e3efe71ac3e452bf79725e8c).